src/store.ts

```diff
@@ -461,6 +467,27 @@ export default function Store(
         mpInstance._Persistence.update();
     };
 
+    this.addMpidToSessionHistory = (mpid: MPID, previousMPID?: MPID): void => {
+        const indexOfMPID = this.currentSessionMPIDs.indexOf(mpid);
+
+        if (
+            // QUESTION: Is sessionId necessary here?
+            this.sessionId &&
+            mpid &&
+            previousMPID !== mpid &&
+            indexOfMPID < 0
+        ) {
+            this.currentSessionMPIDs.push(mpid);
```

Comment:
to make this function more readable in that it will only ever hit one of these, maybe adding a return here would help

```suggestion
            this.currentSessionMPIDs.push(mpid);
            return;
```